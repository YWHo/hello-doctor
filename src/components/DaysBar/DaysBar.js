import React from 'react';
import styled from 'styled-components';
import moment from 'moment';
import * as dateHelper from '../../shared/dateHelper';

const Container = styled.div`
  display: block;
  height: 82px;
  padding: 8px 2px;
  overflow-x: auto;
  overflow-y: hidden;
  white-space: nowrap;
`;

const ButtonCircle = styled.button`
  background-color: ${props => (props.selected ? '#177d91' : 'transparent')};
  border: none;
  border-radius: 50%;
  color: #fff;
  height: 66px;
  width: 66px;

  :focus {
    outline: none;
  }

  :active {
    outline: none;
  }

  :hover {
    background-color: #20aac5;
  }
`;

const DateNum = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 27px;
  font-size: 24px;
`;

const WeekDayName = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 16px;
  font-size: 14px;
  font-weight: 300;
`;

export default function DaysBar(props) {
  const { selectedDate = moment() } = props;

  const days = dateHelper.getDatesInThreeMonthsFrom();
  console.log('days: ', days);
  console.log('selectedDate', moment(selectedDate).format('YYYY-MM-DD'));
  const dateList = days.map((day, idx) => {
    const tmpDate = moment(day);
    const selected = day === moment(selectedDate).format('YYYY-MM-DD');
    return (
      <ButtonCircle key={`dateCi_${idx}`} selected={selected}>
        <DateNum key={`dateNu_${idx}`}>{tmpDate.format('D')}</DateNum>
        <WeekDayName key={`weekDa_${idx}`}>
          {tmpDate.format('ddd').toLowerCase()}
        </WeekDayName>
      </ButtonCircle>
    );
  });

  return <Container>{dateList}</Container>;
}
