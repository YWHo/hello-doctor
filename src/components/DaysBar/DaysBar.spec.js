import React from 'react';
import dayjs from 'dayjs';
import { DaysBar } from './DaysBar';
import renderer from 'react-test-renderer';

describe('NavBar', () => {
  it('render correctly', () => {
    const component = renderer.create(
      <DaysBar selectedDate={dayjs('2019-06-19')} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
