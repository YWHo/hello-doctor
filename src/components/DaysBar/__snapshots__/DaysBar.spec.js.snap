// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render correctly 1`] = `
<div
  className="sc-bdVaJa jcfMkX"
>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      18
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH fIMks"
    selected={true}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      19
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      20
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      21
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      22
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      23
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      24
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      25
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      26
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      27
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      28
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      29
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      30
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      1
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      2
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      3
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      4
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      5
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      6
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      7
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      8
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      9
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      10
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      11
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      12
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      13
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      14
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      15
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      16
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      17
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      18
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      19
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      20
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      21
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      22
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      23
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      24
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      25
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      26
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      27
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      28
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      29
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      30
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      31
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      1
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      2
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      3
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      4
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      5
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      6
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      7
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      8
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      9
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      10
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      11
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      12
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      13
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      14
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      15
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      16
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      17
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      18
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      19
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      20
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      21
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      22
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      23
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      24
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      25
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sun
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      26
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      mon
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      27
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      tue
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      28
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      wed
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      29
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      thu
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      30
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      fri
    </div>
  </button>
  <button
    className="sc-bwzfXH cspXqa"
    selected={false}
  >
    <div
      className="sc-htpNat eeOEiB"
    >
      31
    </div>
    <div
      className="sc-bxivhb gtZwxw"
    >
      sat
    </div>
  </button>
</div>
`;
